engine/table/src/main/java/io/deephaven/engine/table/impl/chunkfilter/ByteChunkMatchFilterFactory.java

```diff
@@ -56,16 +56,79 @@ private SingleValueByteChunkFilter(byte value) {
             this.value = value;
         }
 
+        private boolean matches(byte value) {
+            return value == this.value;
+        }
+
+        /*
+         * The following functions are identical and repeated for each of the filter types. This is to aid the JVM in
+         * correctly inlining the matches() function. The goal is to have a single virtual call per chunk rather than
+         * once per value. This improves performance on JVM <= 21, but may be unnecessary on newer JVMs.
+         */
         @Override
-        public void filter(ByteChunk<? extends Values> values, LongChunk<OrderedRowKeys> keys,
-                WritableLongChunk<OrderedRowKeys> results) {
+        public void filter(
+                final Chunk<? extends Values> values,
+                final LongChunk<OrderedRowKeys> keys,
+                final WritableLongChunk<OrderedRowKeys> results) {
+            final ByteChunk<? extends Values> typedChunk = values.asByteChunk();
+            final int len = typedChunk.size();
+
             results.setSize(0);
-            for (int ii = 0; ii < values.size(); ++ii) {
-                if (values.get(ii) == value) {
+            for (int ii = 0; ii < len; ++ii) {
+                if (matches(typedChunk.get(ii))) {
                     results.add(keys.get(ii));
                 }
             }
         }
+
+        @Override
+        public int filter(final Chunk<? extends Values> values, final WritableBooleanChunk<Values> results) {
+            final ByteChunk<? extends Values> typedChunk = values.asByteChunk();
+            final int len = values.size();
+            int count = 0;
+            for (int ii = 0; ii < len; ++ii) {
+                final boolean newResult = matches(typedChunk.get(ii));
+                results.set(ii, newResult);
+                count += newResult ? 1 : 0;
+            }
+            return count;
+        }
+
+        @Override
+        public int filterAnd(final Chunk<? extends Values> values, final WritableBooleanChunk<Values> results) {
+            final ByteChunk<? extends Values> typedChunk = values.asByteChunk();
+            final int len = values.size();
+            int count = 0;
+            // Count the values that changed from true to false
+            for (int ii = 0; ii < len; ++ii) {
+                final boolean result = results.get(ii);
+                if (!result) {
+                    continue; // already false, no need to compute
+                }
+                boolean newResult = matches(typedChunk.get(ii));
+                results.set(ii, newResult);
+                count += newResult ? 0 : 1;
```

Comment:
I think we should make the return value of filterAnd match filter; and return the count of matched values.  The number of things we set to false seems more complex and error prone in general usage, although it maps fairly well to your use case in the count where operator - we know the existing count because it is used in sequence.